packages/venia-ui/lib/components/FormError/formError.js

```diff
@@ -20,9 +40,9 @@ const FormError = props => {
 
     const classes = useStyle(defaultClasses, propClasses);
 
-    return errorMessage ? (
+    return errorMessage || graphQLErrorMessage() ? (
```

Comment:
Since `errorMessage` already comes from the talon (and talons are where we try to keep component "logic" like this), I'd like to see the `graphQLErrorMessage` function moved to the talon.

This way the component can simply ask the talon for the `errorMessage` and display it (if present).

Looking into `useFormError` a bit deeper, it seems like we may be able to drop its usage of [deriveErrorMessage](https://github.com/magento/pwa-studio/blob/develop/packages/peregrine/lib/util/deriveErrorMessage.js#L18), and replace it with `graphQLErrorMessage` instead.

Reply:
Updated